pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -1168,6 +1168,98 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
+				By("Waiting for VAP to be ready")
+				machineVap = &admissionregistrationv1.ValidatingAdmissionPolicy{}
+				Eventually(k8sClient.Get(ctx, client.ObjectKey{Name: "openshift-only-create-mapi-machine-if-authoritative-api-capi"}, machineVap), timeout).Should(Succeed())
+				Eventually(k.Update(machineVap, func() {
+					machineVap.Spec.Validations = append(machineVap.Spec.Validations, admissionregistrationv1.Validation{
+						Expression: "!(has(object.metadata.labels) && \"test-sentinel\" in object.metadata.labels)",
+						Message:    "policy in place",
+					})
+					// Updating the VAP so that it functions on "UPDATE" only in this test suite to make it easier to test the functionality
+					machineVap.Spec.MatchConstraints.ResourceRules = append(machineVap.Spec.MatchConstraints.ResourceRules, admissionregistrationv1.NamedRuleWithOperations{
+						RuleWithOperations: admissionregistrationv1.RuleWithOperations{
+							Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create, admissionregistrationv1.Update},
+							Rule: admissionregistrationv1.Rule{
+								APIGroups:   []string{"machine.openshift.io"},
+								APIVersions: []string{"*"},
+								Resources:   []string{"machines"},
+							},
+						},
+					})
+
+				})).Should(Succeed())
+
+				Eventually(k.Object(machineVap), timeout).Should(
+					HaveField("Status.ObservedGeneration", BeNumerically(">=", 2)),
+				)
+
+				By("Updating the VAP binding")
+				policyBinding = &admissionregistrationv1.ValidatingAdmissionPolicyBinding{}
+				Eventually(k8sClient.Get(ctx, client.ObjectKey{
+					Name: "openshift-only-create-mapi-machine-if-authoritative-api-capi"}, policyBinding), timeout).Should(Succeed())
+
+				Eventually(k.Update(policyBinding, func() {
+					// We want to have our paramref reference the CAPI namespace,
+					// since we `GenerateName` it is not static
+					policyBinding.Spec.ParamRef.Namespace = capiNamespace.GetName()
+					// We need to update the namespace in our namespaceSelector,
+					// since also use `GenerateName` here
+					policyBinding.Spec.MatchResources.NamespaceSelector.MatchLabels = map[string]string{
+						"kubernetes.io/metadata.name": mapiNamespace.GetName(),
+					}
+				}), timeout).Should(Succeed())
+
+				// Wait until the binding shows the patched values
+				Eventually(k.Object(policyBinding), timeout).Should(
+					SatisfyAll(
+						HaveField("Spec.MatchResources.NamespaceSelector.MatchLabels",
+							HaveKeyWithValue("kubernetes.io/metadata.name",
+								mapiNamespace.GetName())),
+					),
+				)
+
+				checkVAPCapiMachine := clusterv1resourcebuilder.Machine().WithGenerateName("capi-vap-checking-machine").WithNamespace(capiNamespace.Name).Build()
+				Expect(k8sClient.Create(ctx, checkVAPCapiMachine)).To(Succeed())
+
+				Eventually(k.Get(checkVAPCapiMachine)).Should(Succeed())
+
+				By("Creating a throwaway MAPI machine")
+				testMachine := mapiMachineBuilder.WithGenerateName("test-machine").Build()
```

Comment:
This name doesn't match our CAPI machine created above too. We are creating a CAPI machine somewhere with the same name then 🤔